packages/wonder-blocks-modal/components/modal-launcher-portal.test.js

```diff
@@ -6,41 +6,40 @@ import ModalLauncherPortal from "./modal-launcher-portal.js";
 import {ModalLauncherPortalAttributeName} from "../util/constants.js";
 
 describe("ModalPortal", () => {
-    test("Mounts its children in document.body", (done) => {
+    test("Mounts its children in document.body", async () => {
+        const element = await new Promise((resolve, reject) => {
+            // We include an extra wrapper element here, just to extra confirm
+            // that this _isn't_ part of the tree where the children get
+            // mounted.
+            mount(
+                <div data-this-should-not-contain-the-children>
+                    <ModalLauncherPortal>
+                        <div ref={resolve} />
+                    </ModalLauncherPortal>
+                </div>,
+            );
+        });
+
         // Once the child mounts, check that it was mounted directly-ish into
         // `document.body`, and finish the test.
-        const childrenRef = (element) => {
-            if (element) {
-                // Find the nearest parent, disregarding nodes created by
-                // ModalLauncherPortal and by `ReactDOM.render`.
-                let parent = element.parentElement;
-                while (
-                    parent &&
-                    (parent.hasAttribute(ModalLauncherPortalAttributeName) ||
-                        parent.hasAttribute("data-reactroot"))
-                ) {
-                    parent = parent.parentNode;
-                }
-
-                // This nearest parent _should_ be document.body. It definitely
-                // should not be the `data-this-should-not-contain-the-children`
-                // element! That element shouldn't be in the ancestor chain at
-                // all.
-                expect(parent).toBe(document.body);
-                done();
+        if (element) {
```

Comment:
@matchu if there is no `element` should the test fail?

Reply:
I think this was because we're effectively treating this as a `componentDidMount` handler, but `ref` is called on both mount and unmount, and we want to ignore the unmount case. Probably would be better to short-circuit if `element` is absent, with a comment explaining why. Sorry for the ambiguity!

Reply:
That makes sense.  I keep forgetting that it's called on unmount as well.